src/site/siteSlice.ts

```diff
@@ -82,6 +86,20 @@ export const deleteSite = createAsyncThunk<string, Site>(
   },
 );
 
+export const transferSites = createAsyncThunk<
+  Awaited<ReturnType<typeof siteService.transferSitesToProject>>,
+  SiteTransferMutationInput
+>('site/transferSites', async (input, _currentUser, { dispatch }) => {
+  const result = await siteService.transferSitesToProject(input);
+  for (const { siteId, oldProjectId } of result.updated) {
+    if (oldProjectId !== undefined) {
```

Comment:
question: do you need to check for undefined? or any true value? or does TypeScript ensure it can't be false/null/""?

Reply:
Oh yes, the type is `string | undefined` here.